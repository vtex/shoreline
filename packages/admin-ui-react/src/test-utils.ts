import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
