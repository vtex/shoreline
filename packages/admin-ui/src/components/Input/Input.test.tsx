import React from 'react'
import { render, axe } from '../../test-utils'
import { IconPlus } from '@vtex/phosphor-icons'

import { Input } from './index'

describe('Input tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <>
        <Input
          charLimit={120}
          helperText="Helper Text"
          value=""
          onChange={() => {}}
          label="Label"
          id="text-field-1"
        />
        <Input
          charLimit={120}
          helperText="Helper Text"
          criticalText="Error Message"
          value=""
          onChange={() => {}}
          label="Label"
          id="text-field-2"
        />
        <Input
          icon={<IconPlus />}
          suffix="Kg"
          charLimit={120}
          helperText="Helper Text"
          value=""
          onChange={() => {}}
          onClear={() => {}}
          label="Label"
          id="text-field-3"
        />
        <Input
          icon={<IconPlus />}
          suffix="Kg"
          charLimit={120}
          helperText="Helper Text"
          criticalText="Error Message"
          value=""
          onChange={() => {}}
          onClear={() => {}}
          label="Label"
          id="text-field-4"
        />
      </>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
