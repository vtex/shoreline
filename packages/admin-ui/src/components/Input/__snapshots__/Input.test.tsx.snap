// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 input:focus+label,
.emotion-3 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-3 input:placeholder-shown:not(:focus)+label,
.emotion-3 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-3 input:not(:placeholder-shown)+label,
.emotion-3 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  position: relative;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 0.75rem;
  padding-right: 1rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-0:hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
}

<div
    class="emotion-3"
  >
    <div
      class="emotion-2"
    >
      <input
        class="emotion-0"
        id="text-field-1"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-1"
        for="text-field-1"
      >
        Label
      </label>
    </div>
  </div>
  .emotion-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 input:focus+label,
.emotion-5 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-5 input:placeholder-shown:not(:focus)+label,
.emotion-5 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-5 input:not(:placeholder-shown)+label,
.emotion-5 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  position: relative;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 0.75rem;
  padding-right: 1rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-0:hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
}

<div
    class="emotion-5"
  >
    <div
      class="emotion-2"
    >
      <input
        class="emotion-0"
        id="text-field-2"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-1"
        for="text-field-2"
      >
        Label
      </label>
    </div>
    <div
      class="emotion-4"
    >
      <span
        class="emotion-3"
      >
        Helper Text
      </span>
    </div>
  </div>
  .emotion-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 input:focus+label,
.emotion-5 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-5 input:placeholder-shown:not(:focus)+label,
.emotion-5 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-5 input:not(:placeholder-shown)+label,
.emotion-5 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  position: relative;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 0.75rem;
  padding-right: 1rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-0:hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
}

<div
    class="emotion-5"
  >
    <div
      class="emotion-2"
    >
      <input
        class="emotion-0"
        id="text-field-3"
        maxlength="120"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-1"
        for="text-field-3"
      >
        Label
      </label>
    </div>
    <div
      class="emotion-4"
    >
      <div />
      <span
        class="emotion-3"
      >
        0/120
      </span>
    </div>
  </div>
  .emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6 input:focus+label,
.emotion-6 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-6 input:placeholder-shown:not(:focus)+label,
.emotion-6 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-6 input:not(:placeholder-shown)+label,
.emotion-6 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  position: relative;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 0.75rem;
  padding-right: 1rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-0:hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-2"
    >
      <input
        class="emotion-0"
        id="text-field-4"
        maxlength="120"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-1"
        for="text-field-4"
      >
        Label
      </label>
    </div>
    <div
      class="emotion-5"
    >
      <span
        class="emotion-3"
      >
        Helper Text
      </span>
      <span
        class="emotion-3"
      >
        0/120
      </span>
    </div>
  </div>
  .emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6 input:focus+label,
.emotion-6 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-6 input:placeholder-shown:not(:focus)+label,
.emotion-6 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-6 input:not(:placeholder-shown)+label,
.emotion-6 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  position: relative;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 0.75rem;
  padding-right: 1rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-0:hover {
  border: 1px solid #545454;
}

.emotion-0:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:disabled {
  background: #E6E6E6;
  color: #747474;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-2"
    >
      <input
        class="emotion-0"
        id="text-field-5"
        maxlength="120"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-1"
        for="text-field-5"
      >
        Label
      </label>
    </div>
    <div
      class="emotion-5"
    >
      <span
        class="emotion-3"
      >
        Helper Text
      </span>
      <span
        class="emotion-3"
      >
        0/120
      </span>
    </div>
  </div>
  .emotion-9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-9 input:focus+label,
.emotion-9 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-9 input:placeholder-shown:not(:focus)+label,
.emotion-9 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-9 input:not(:placeholder-shown)+label,
.emotion-9 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-6 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 44px;
  padding-right: 75px;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-1:hover {
  border: 1px solid #545454;
}

.emotion-1:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-1:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-5 {
  position: relative;
}

.emotion-5 label {
  left: 44px;
}

.emotion-0 {
  height: 20px;
  width: 20px;
  min-height: 20px;
  min-width: 20px;
  color: #545454;
  top: 0.25rem;
  left: 0rem;
  margin-left: 0.75rem;
  margin-right: 0.75rem;
  margin-top: 14px;
  position: absolute;
}

.emotion-4 {
  right: 0rem;
  top: 0.25rem;
  height: 46px;
  padding-right: 0.75rem;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
}

.emotion-3 {
  color: #000000;
  border-left: 1px solid #B9B9B9;
  padding-top: 14px;
  margin-top: 0.0625rem;
  width: 32px;
  padding-left: 0.75rem;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
}

<div
    class="emotion-9"
  >
    <div
      class="emotion-5"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
          x1="40"
          x2="216"
          y1="128"
          y2="128"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
          x1="128"
          x2="128"
          y1="40"
          y2="216"
        />
      </svg>
      <input
        class="emotion-1"
        id="text-field-6"
        maxlength="120"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-2"
        for="text-field-6"
      >
        Label
      </label>
      <div
        class="emotion-4"
      >
        <span
          class="emotion-3"
        >
          Kg
        </span>
      </div>
    </div>
    <div
      class="emotion-8"
    >
      <span
        class="emotion-6"
      >
        Helper Text
      </span>
      <span
        class="emotion-6"
      >
        0/120
      </span>
    </div>
  </div>
  .emotion-9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-9 input:focus+label,
.emotion-9 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-9 input:placeholder-shown:not(:focus)+label,
.emotion-9 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-9 input:not(:placeholder-shown)+label,
.emotion-9 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  left: 12px;
  padding-top: 0.5rem;
  color: #545454;
  margin-bottom: 0.75rem;
  position: absolute;
  -webkit-transform: translate(0, 16px) scale(1);
  -moz-transform: translate(0, 16px) scale(1);
  -ms-transform: translate(0, 16px) scale(1);
  transform: translate(0, 16px) scale(1);
  transform-origin: top left;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.25rem;
}

.emotion-6 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  width: 100%;
  height: 48px;
  padding-left: 44px;
  padding-right: 75px;
  background: transparent;
  border: 1px solid #B9B9B9;
  border-radius: 4px;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  color: #000000;
  outline: 0;
  -webkit-transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  transition: all 70ms cubic-bezier(0.2, 0.2, 0.38, 0.9);
  padding-top: 1rem;
}

.emotion-1:hover {
  border: 1px solid #545454;
}

.emotion-1:focus {
  border: 1px solid #3B3B3B;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-1:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-5 {
  position: relative;
}

.emotion-5 label {
  left: 44px;
}

.emotion-0 {
  height: 20px;
  width: 20px;
  min-height: 20px;
  min-width: 20px;
  color: #545454;
  top: 0.25rem;
  left: 0rem;
  margin-left: 0.75rem;
  margin-right: 0.75rem;
  margin-top: 14px;
  position: absolute;
}

.emotion-4 {
  right: 0rem;
  top: 0.25rem;
  height: 46px;
  padding-right: 0.75rem;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
}

.emotion-3 {
  color: #000000;
  border-left: 1px solid #B9B9B9;
  padding-top: 14px;
  margin-top: 0.0625rem;
  width: 32px;
  padding-left: 0.75rem;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
}

<div
    class="emotion-9"
  >
    <div
      class="emotion-5"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
          x1="40"
          x2="216"
          y1="128"
          y2="128"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
          x1="128"
          x2="128"
          y1="40"
          y2="216"
        />
      </svg>
      <input
        class="emotion-1"
        id="text-field-7"
        maxlength="120"
        placeholder=" "
        value=""
      />
      <label
        class="emotion-2"
        for="text-field-7"
      >
        Label
      </label>
      <div
        class="emotion-4"
      >
        <span
          class="emotion-3"
        >
          Kg
        </span>
      </div>
    </div>
    <div
      class="emotion-8"
    >
      <span
        class="emotion-6"
      >
        Helper Text
      </span>
      <span
        class="emotion-6"
      >
        0/120
      </span>
    </div>
  </div>
</DocumentFragment>
`;
